src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,128 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
```

Comment:
Ensure that `ArrayListIndexOutOfBoundsException` is defined within your project, as it is used in the code but not provided in the current context.